web/ui/react-app/src/pages/TSDBStatus.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { FC, Fragment } from 'react';
+import { RouteComponentProps } from '@reach/router';
+import { Alert, Table, Badge } from 'reactstrap';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faSpinner } from '@fortawesome/free-solid-svg-icons';
+import { useFetch } from '../utils/useFetch';
+import PathPrefixProps from '../PathPrefixProps';
+
+export interface Stats {
+  name: string;
+  value: number;
+}
+
+export interface TSDBMap {
+  queryTime: string;
+  scrapeValueCountByMetricsName: Array<Stats>;
+  labelValueCountByLabelName: Array<Stats>;
+  memoryUsageInBytesByLabelName: Array<Stats>;
+  scrapeValueCountByLabelValuePair: Array<Stats>;
+}
+
+const paddingStyle = {
+  padding: '10px',
+};
+
+function createTable(title: string, unit: string, stats: Array<Stats>) {
+  return (
+    <div style={paddingStyle}>
+      <h3>{title}</h3>
+      <Table bordered={true} size="sm" striped={true}>
+        <thead>
+          <tr>
+            <th>Name</th>
+            <th>{unit}</th>
+          </tr>
+        </thead>
+        <tbody>
+          {stats.map((element: Stats, i: number) => {
+            return (
+              <Fragment key={i}>
+                <tr>
+                  <td>{element.name}</td>
+                  <td>{element.value}</td>
+                </tr>
+              </Fragment>
+            );
+          })}
+        </tbody>
+      </Table>
+    </div>
+  );
+}
+
+const TSDBStatus: FC<RouteComponentProps & PathPrefixProps> = ({ pathPrefix }) => {
+  const { response, error } = useFetch(`${pathPrefix}/api/v1/status/tsdb`);
+  const headStats = () => {
+    const stats: TSDBMap = response && response.data;
+    if (error) {
+      return (
+        <Alert color="danger">
+          <strong>Error:</strong> Error fetching TSDB Status: {error.message}
+        </Alert>
+      );
+    } else if (stats) {
+      let color = 'success';
+      const qTime = Number(stats.queryTime);
+      if (qTime >= 1) {
+        color = 'warning';
+      }
+
+      return (
+        <div>
+          <div style={paddingStyle}>
+            <h3>Head Cardinality Stats</h3>
+            <div>
+              Query Time <Badge color={color}>{stats.queryTime}</Badge> Seconds
+            </div>
+          </div>
+          {createTable('Top 10 Lable Names With Value Count', 'Count', stats.labelValueCountByLabelName)}
+          {createTable('Top 10 Metric Names With High Scrap Value Count ', 'Count', stats.scrapeValueCountByMetricsName)}
+          {createTable('Top 10 Label Names With High Memory Usage', 'Bytes', stats.memoryUsageInBytesByLabelName)}
+          {createTable('Top 10 Label Pairs with Scrap Value Count', 'Count', stats.scrapeValueCountByLabelValuePair)}
```

Comment:
Scrap -> Scrape

Reply:
Whoops, though everywhere where I corrected "scrape" in some way, I guess the names will change anyway due to my other comment about field naming.